hbase-server/src/main/java/org/apache/hadoop/hbase/master/locking/LockManager.java

```diff
@@ -108,19 +108,10 @@ public MasterLock(final RegionInfo[] regionInfos, final String description) {
       this.description = description;
     }
 
-    /**
-     * Acquire the lock, waiting indefinitely until the lock is released or the thread is
-     * interrupted.
-     * @throws InterruptedException If current thread is interrupted while waiting for the lock
-     */
-    public boolean acquire() throws InterruptedException {
-      return tryAcquire(0);
-    }
-
     /**
      * Acquire the lock within a wait time.
-     * @param timeoutMs The maximum time (in milliseconds) to wait for the lock, 0 to wait
```

Comment:
Why we need to introduce this change?

Reply:
It seems reasonable to have this change, while it is only being used by snapshot as of today. If it is 0, client indicates that it has 0ms as timeout. Does that not sound right?

Reply:
If client would like infinite timeout, it better provide Long.MAX_VALUE or negative value. But having 0ms timeout is also a valid case? (at least from testing viewpoint)

Reply:
In your current implementation, if you set timeout 0, it is possible that we fail before entering the while loop, if the return value of EnvironmentEdgeManager.currentTime changes when testing the condition of while loop...

And could you please check all the references to this method? We need to to know if this introduces any troubles or breaks our compatibility guarantee...

Reply:
That is the intended behaviour with 0ms timeout (specifically for tests).
And yes, this is the only usecase of this method because this is specific way of creating LockProc by using CountdownLatch.

Reply:
I do not think we can rely this in tests? It is unstable, if the statement execute fast enough then we will enter the loop, if not we will not enter...

Reply:
Ok then let me introduce some custom utility to wait before entering loop, just for testing purpose.

Reply:
This is taken care of.